XenAdmin/Dialogs/CallHome/HealthCheckOverviewLauncher.cs

```diff
@@ -0,0 +1,193 @@
+{
+    public class HealthCheckOverviewLauncher
+    {
+        private static readonly log4net.ILog log = log4net.LogManager.GetLogger(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);
+        private bool healthCheckOverviewVisible;
+        private CallHomeOverviewDialog healthCheckOverviewDialog;
+        private readonly IWin32Window parent;
+        private DateTime LastCloseTime { get; set; }
+        private readonly object healthCheckLock = new object();
+
+        protected virtual DateTime ReferenceTime
+        {
+            get { return DateTime.Now; }
+        }
+
+        protected virtual TimeSpan TimeSinceLastClose
+        {
+            get { return ReferenceTime - LastCloseTime; }
+        }
+
+        protected virtual bool ModalDialogVisible
+        {
+            get { return Win32Window.ModalDialogIsVisible(); }
+        }
+
+        protected virtual DialogResult LaunchDialog(IEnumerable<IXenObject> selectedObjects)
+        {
+            if (healthCheckOverviewDialog == null)
+                return DialogResult.None;
+
+            return healthCheckOverviewDialog.ShowDialog(parent, selectedObjects.ToList());
+        }
+
+        protected virtual void RefreshDialog(IEnumerable<IXenObject> selectedObjects)
+        {
+            if (healthCheckOverviewDialog == null)
+                return;
+
+            healthCheckOverviewDialog.RefreshView(selectedObjects.ToList());
+        }
+
+        public HealthCheckOverviewLauncher(IWin32Window parent)
+        {
+            parent = parent;
+            healthCheckOverviewVisible = false;
+        }
+
+        public bool HealthCheckDialogIsShowing
+        {
+            get { return healthCheckOverviewDialog != null; }
+        }
+
+        private void LoadHealthCheckOverviewDialog()
+        {
+            healthCheckOverviewDialog = new CallHomeOverviewDialog();
+        }
+
+        public void LaunchIfRequired(bool nag, SelectedItemCollection selectedObjects)
+        {
+            if (selectedObjects != null && selectedObjects.AllItemsAre<IXenObject>(x => x is Pool || x is Host))
+            {
+                List<IXenObject> itemsSelected = selectedObjects.AsXenObjects<Pool>().ConvertAll(p => p as IXenObject);
+                itemsSelected.AddRange(selectedObjects.AsXenObjects<Host>().Select
+                    (host => Helpers.GetPoolOfOne(((IXenObject)host).Connection)).Cast<IXenObject>().Distinct());
+
+                LaunchIfRequired(nag, itemsSelected);
+            }
+            else
+                LaunchIfRequired(nag, new List<IXenObject>());
+        }
+
+        private void LaunchIfRequired(bool nag, IEnumerable<IXenObject> selectedObjects)
+        {
+            lock (healthCheckLock)
+            {
+                if (!healthCheckOverviewVisible)
+                {
+                    LoadHealthCheckOverviewDialog();
+                    if (nag && TimeSinceLastClose < TimeSpan.FromSeconds(10))
```

Comment:
I am just thinking about a case when connecting to multiple hosts would take minutes, that would mean too many pop ups that have to be clicked - some would find it annoying (they are modal...).